### PR TITLE
Add Model Evaluation Panel

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

- Add a new build-in Model Evaluation panel 

## How is this patch tested? If it is not, please explain why.

Using the panel in the app with several evaluations

## Todos

- [x] Include evaluation model operator
- [x] Add interactivity to plots and table
- [x] Beta badge
- [x] Histogram bins configuration modal
- [x] Test with Classification, Segmentation
- [x] Switch to using evaluation doc id instead of key
- [x] Use store to cache evaluation data needed for panel
- [x] Add confidence metrics
- [x] Hide plotly action
- [x] Markdown notes
- [x] Expandable notes
- [x] Add per-class IoU
- [x] Permission
- [x] FiftyOne Teams support 

## Todos (Extras)

- [x] Add computing evaluation card
- [x] Add queued evaluation card
- [ ] ~Lazy loading data as card is expanded and the view is updated~

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced `ThemeProvider` updates for enhanced theme customization, including a new `card` color in the dark theme and a new button variant with white text.
	- Added several new components for model evaluation, including `EmptyOverview`, `Evaluate`, `Evaluation`, `EvaluationNotes`, `EvaluationPlot`, `EvaluationTable`, `Overview`, and `Status`.
	- Implemented the `EvaluationPanel` class for comprehensive model evaluation capabilities, including metrics retrieval and visualization.

- **Bug Fixes**
	- Adjusted z-index values for `MuiModal` and `MuiMenu` components to improve UI layering.

- **Chores**
	- Updated utility functions for event handling and numerical calculations to enhance functionality.
	- Enhanced the `isCompositeView` function to support additional schema properties.

### CODE REVIEW

File: app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx [hunk 1]
```diff
@@ -0,0 +1,1327 @@
...
+  const { can_edit_note, can_edit_status } = data?.permissions || {};
+
+  useEffect(() => {
+    if (!evaluation) {
+      loadEvaluation();
+    }
+  }, [evaluation]);
+
+  useEffect(() => {
+    if (!compareEvaluation && !loadingCompare && compareKey) {
```
Comment (manivoxel51) [inline app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx:121]: 🥶

File: app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx [hunk 2]
```diff
@@ -0,0 +1,1303 @@
...
+                                )}
+                              </Stack>
+                            </TableCell>
+                            <TableCell>
+                              <Stack
+                                direction="row"
+                                sx={{ justifyContent: "space-between" }}
+                              >
+                                <Typography>{difference}</Typography>
+                                {!isNaN(ratio) && (
```
[the remark this answers is not in the document]
Comment (manivoxel51) [reply app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx:684]: might be a good idea

File: app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx [hunk 3]
```diff
@@ -0,0 +1,1327 @@
...
+  }, [evaluation]);
+  const compareConfusionMatrix = useMemo(() => {
+    return compareEvaluation?.confusion_matrix;
+  }, [compareEvaluation]);
+  const compareKeys = useMemo(() => {
+    const keys: string[] = [];
+    const evaluations = data?.evaluations || [];
+    for (const evaluation of evaluations) {
+      const { key } = evaluation;
+      if (key !== name) {
```
Comment (manivoxel51) [inline app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/Evaluation.tsx:100]: 👍

File: app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/EvaluationNotes.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    }
+  }, [notesRef]);
+
+  useLayoutEffect(() => {
+    truncate();
+  }, [truncate]);
+
+  useEffect(() => {
+    truncate();
+  }, [notes, truncate]);
```
[the remark this answers is not in the document]
Comment (manivoxel51) [reply app/packages/core/src/plugins/SchemaIO/components/NativeModelEvaluationView/EvaluationNotes.tsx:34]: maybe no need for useEffect here?

**Review by manivoxel51 — APPROVED** [review]
code Looking great! 🍨 

I need to sit with you next and set it up to test the functionality more

File: fiftyone/operators/builtins/panels/model_evaluation/__init__.py [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+class EvaluationPanel(Panel):
+    @property
+    def config(self):
+        return PanelConfig(
+            name="evaluation_panel",
+            label="Evaluation Panel",
+            icon="ssid_chart",
+            category=Categories.ANALYZE,
+            beta=True,
+            is_new=is_new("2024-11-07"),
```
Comment (Br2850) [inline fiftyone/operators/builtins/panels/model_evaluation/__init__.py:30]:
```suggestion
            is_new=is_new("2024-11-14"),
```

The 14th is when all these panels regardless of OSS & Teams are talked about publicly so let's set that date (that's what we did for Model Eval)

**Pull request merged**